jupyterlab_git/handlers.py

```diff
@@ -27,6 +27,36 @@ class GitHandler(APIHandler):
     def git(self):
         return self.settings["git"]
 
+    def git_error_no_500(self, result, log_lvl="info"):
+        """
+        Check git error message output and return True if found in `error_message` list.
+        If found, error message is displayed in terminal as log output (default "info" level).
+        Inputs:
+            `result` is output of Git.show_top_level:
+                {
+                    "code": _assumes_non-zero_code_,
+                    "command": "git rev-parse --show-toplevel",
+                    "message": "fatal: not a git repository (or any of the parent directories):..",
+                }
+            `log_lvl`: str = "info" (can be "warning" or "error")
+        """
+        # Error messages from git to handle via logger (info) instead of 500 error
+        error_message = map(
+            lambda n: n.lower(),
+            [
+                "fatal: not a git repository (or any of the parent directories):",
+            ]
+        )
+        lower_error = result.get("message", "").lower()
+        if any([msg in lower_error for msg in error_message]):
+            log_lvl = log_lvl if log_lvl in ["info", "warning", "error"] else "info"
+            getattr(self.log, log_lvl)(
+                "[jupyterlab_git] Failed to execute 'git' command: {!s},"
+                " with message: {!s}".format(result["command"], result["message"].strip())
+            )
+            return True
+        return False
```

Comment:
Could you move this after the command execution https://github.com/jupyterlab/jupyterlab-git/blob/90cfb96d5f029badf85e834b5573f9871c85b58f/jupyterlab_git/git.py#L682

Then if the error is the expected, set the returned value to:

```
{"code": code, "top_repo_path": None}
```

Could you send the error message on debug level? Basically we try to be as quiet as possible in default mode.

Reply:
Done in new commit 0cdba5b: returns code 0 with top_repo_path None